### PR TITLE
Add triggerer info to task instance in API

### PR DESCRIPTION
**Pull request by dstandish**
Showing triggerer info in a section at the top of grid view TI details section
section will only be visible when ti state is deferred
image

### CODE REVIEW

**Review by uranusjr — APPROVED** [review]

**Comment by uranusjr** [comment]
Code makes sense to me. A few tests on the API and this should be good to go.

File: airflow/example_dags/example_time_delta_sensor_async.py [hunk 1]
```diff
@@ -36,6 +36,6 @@
     catchup=False,
     tags=["example"],
 ) as dag:
-    wait = TimeDeltaSensorAsync(task_id="wait", delta=datetime.timedelta(seconds=10))
+    wait = TimeDeltaSensorAsync(task_id="wait", delta=datetime.timedelta(seconds=120))
```
Comment (jedcunningham) [inline airflow/example_dags/example_time_delta_sensor_async.py]:
```suggestion
    wait = TimeDeltaSensorAsync(task_id="wait", delta=datetime.timedelta(seconds=10))
```

Pretty big jump in runtime for an example dag... Maybe leave it as it was, or a more modest increase instead?

File: airflow/models/taskinstance.py [hunk 2]
```diff
@@ -509,14 +509,8 @@ class TaskInstance(Base, LoggingMixin):
...
 
-    trigger = relationship(
-        "Trigger",
-        primaryjoin="TaskInstance.trigger_id == Trigger.id",
-        foreign_keys=trigger_id,
-        uselist=False,
-        innerjoin=True,
-    )
-
+    trigger = relationship("Trigger", uselist=False)
```
Comment (jedcunningham) [inline airflow/models/taskinstance.py:509]: Why don't we need innerjoin any longer?

File: airflow/example_dags/example_time_delta_sensor_async.py (hunk 1 above)
Comment (dstandish) [reply airflow/example_dags/example_time_delta_sensor_async.py]: was an accident, thanks

File: airflow/models/taskinstance.py (hunk 2 above)
Comment (dstandish) [reply airflow/models/taskinstance.py:509]: i don't think we ever needed it.  not every TI has a trigger, so what happens is... when you reference this property in a query, the TIs that don't have a trigger drop out of the result set.
i looked, and it seems that there was only [one usage](https://github.com/apache/airflow/blob/aa63eb969320cacadaed04fd54c716f57da7a2c7/airflow/www/views.py#L1557) of this attr, and in it, they actually overrided that setting, to set it to false...

**Review by jedcunningham — APPROVED** [review]

**Review by blag — APPROVED** [review]

**Pull request merged**